src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -129,7 +129,9 @@ bool InputStateMachineEngine::_DoControlCharacter(const wchar_t wch, const bool
     if (wch == UNICODE_ETX && !writeAlt)
     {
         // This is Ctrl+C, which is handled specially by the host.
-        success = _pDispatch->WriteCtrlC();
+        const KeyEvent keyDown{ true, 1, 'C', 0, UNICODE_ETX, LEFT_CTRL_PRESSED };
+        const KeyEvent keyUp{ false, 1, 'C', 0, UNICODE_ETX, LEFT_CTRL_PRESSED };

```

Comment:
I feel like with the frequency of this pattern that we should have some sort of static helper on `KeyEvent` that returns a down/up `std::pair` that can be automatically separated with the construction pattern

`const auto [down, up] = KeyEvent::MakePair(1, 'C', 0, UNICODE_ETX, LEFT_CTRL_PRESSED);`